modules/face/src/face_alignment.cpp

```diff
@@ -0,0 +1,344 @@
+warranties of merchantability and fitness for a particular purpose are disclaimed.
+In no event shall copyright holders or contributors be liable for any direct,
+indirect, incidental, special, exemplary, or consequential damages
+(including, but not limited to, procurement of substitute goods or services;
+loss of use, data, or profits; or business interruption) however caused
+and on any theory of liability, whether in contract, strict liability,
+or tort (including negligence or otherwise) arising in any way out of
+the use of this software, even if advised of the possibility of such damage.
+*/
+
+
+/*
+This file contains implementation of the bio-inspired features (BIF) approach
+for computing image descriptors, applicable for human age estimation. For more
+details we refer to [1,2].
+
+REFERENCES
+  [1] Guo, Guodong, et al. "Human age estimation using bio-inspired features."
+      Computer Vision and Pattern Recognition, 2009. CVPR 2009.
+  [2] Spizhevoi, A. S., and A. V. Bovyrin. "Estimating human age using
+      bio-inspired features and the ranking method." Pattern Recognition and
+      Image Analysis 25.3 (2015): 547-552.
+*/
+#include "opencv2/face/face_alignment.hpp"
+#include "opencv2/objdetect.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/imgcodecs.hpp"
+#include "opencv2/core.hpp"
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <map>
+#include <vector>
+#include <cmath>
+using namespace std;
+namespace cv{ 
+namespace face{
+
+const std::string face_cascade_name = "lbpcascade_frontalface_improved.xml";
+const std::string meanshapefile ="/home/sukhad/Desktop/opencv_contrib/modules/face/data/mean_shape.txt";
+
+class ShapePredictor : public cv::face::FaceAlignment{
+public: 
+    virtual int getNumLandmarks() const {return numlandmarks_;}
+
+    virtual int getNumFaces() const {return numfaces_;}
+
+    virtual vector<cv::Point2f> getInitialShape() const { return initialshape_ ;}
+
+
+    virtual std::vector<Point2f> setLandmarks(
+        cv::Mat img,
+        const std::vector<cv::Rect> face,
+        const std::vector< std::vector<cv::Point2f> > landmarks
+    ) const;
+
+
+    ShapePredictor(){
+        ifstream f(meanshapefile);
+        string s;
+        while(getline(f,s)){
+            int i=0;
+            char x1[100],y1[100];
+            while(s[i]!=','){
+                x1[i]=s[i];
+                i++;
+            }
+            string x,y;
+            x=string(x1);
+            i++;
+            int j=0;
+            while(i<s.length()){
+                y1[j++]=s[i];
+                i++;
+            }
+            y=string(y1);
+            Point2f pt;
+            pt.x=std::stof(x);
+            pt.y=std::stof(y);
+            initialshape_.push_back(pt);
+        }
+        landmarks.clear();
+        faces.clear();
+    }
+    ~ShapePredictor(){}
+private:
+    int numlandmarks_;
```

Comment:
I actually went through some code in other files. In them the variables were named like these. So I thought it might be a convention.